RFS/src/test/java/com/rfs/integration/SnapshotStateTest.java

```diff
@@ -0,0 +1,155 @@
+package com.rfs.integration;
+
+import org.apache.lucene.util.IOUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import com.rfs.common.OpenSearchClient;
+import com.rfs.framework.ClusterOperations;
+import com.rfs.framework.ElasticsearchContainer;
+import com.rfs.framework.SimpleRestoreFromSnapshot_ES_7_10;
+import reactor.core.publisher.Mono;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.anyOf;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Tests focused on setting up different snapshot states and then verifying the behavior of RFS towards the target cluster
+ */
+public class SnapshotStateTest {
+
+    private final File SNAPSHOT_DIR = new File("./snapshotRepo");
+    private ElasticsearchContainer cluster;
+    private ClusterOperations operations;
+    private SimpleRestoreFromSnapshot_ES_7_10 srfs;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        // Make sure the snapshot directory is clean before and after tests run
+        IOUtils.rm(Path.of(SNAPSHOT_DIR.getAbsolutePath()));
```

Comment:
This seems mildly unproductive if trying to do differential diagnosis. You could toss a session nonce and delete on success, making the user blow the whole parent directory of runs away

Reply:
Thanks - moved towards your suggestion as the file system bind on docker creating trouble when running as a Github Action.